classtest.js

```diff
@@ -1,87 +1,152 @@
-// console.log(capHill.displayCookies());
-var alki = new CookieStore('Alki:', 2, 16, 4.6);
-// console.log(alki.displayCookies());
-var tableEl = document.createElement('table');
+  //---New store constructors--//
+var pikePlace = new CookieStore ('First and Pike', 23, 65, 6.3);
+var seaTac = new CookieStore ('SeaTac', 3, 24, 1.2);
+var seattleCenter = new CookieStore('Seattle Center', 11, 38, 3.7);
+var capHill = new CookieStore('Capitol Hill', 20, 38, 2.3);
+var alki = new CookieStore('Alki', 2, 16, 4.6);
+  //-- array of the stores for easy access and pulling the info---//
 var stores = [pikePlace, seaTac, seattleCenter, capHill, alki];
+var colTotals = [];
+  //---starts the total at zero so that we can calculate totals within the for loops. resetting to zero allows us to differentiate different rows---//
 var total = 0;
-//----------------------For Loop to run through the times------------
-
-var rowOneEl = document.createElement('tr');
-var anything = document.createElement('th');
-anything.textContent = timeSlot.label;
-rowOneEl.appendChild(anything);
-
-for (var k = 0; k < timeSlot.times.length ; k++){
-  var timeLabelEl = document.createElement('th');
-  var currentTime = timeSlot.times;
-  timeLabelEl.textContent = currentTime[k];
-  rowOneEl.appendChild(timeLabelEl);
-};
-var totalHead = document.createElement('th');
-totalHead.textContent = timeSlot.totalLabel;
-rowOneEl.appendChild(totalHead);
-
-tableEl.appendChild(rowOneEl);
-
+  //---Creates the table element in the html that we insert into-----//
+var tableEl = document.createElement('table');
 document.body.appendChild(tableEl);
 
-for(var i = 0; i < stores.length; i++){
-  var currentStore = stores[i];
+//------------------------FUNCTIONS----------------------------//
+//---------------function to calculate total cookies per hour for each store------
 
-  var rowEl = document.createElement('tr');
+//------------???? what is this again ?????-------------//
 
+// -----------------function to create new cookie store objects.(use of the .this method here)-------//
+function CookieStore(name, minCustomers, maxCustomers, avgCookies) {
+  this.name = name;
+  this.minCustomers = minCustomers;
+  this.maxCustomers = maxCustomers;
+  this.avgCookies = avgCookies;
+  this.hourlySales = [];
+}
+CookieStore.prototype.getAvgCookieCount = function (){
+  var range = this.maxCustomers - this.minCustomers;
+  return Math.floor(Math.random() * range + this.minCustomers);
+};
+CookieStore.prototype.renderRow = function() {
+  var rowEl = document.createElement('tr');
   var nameEl = document.createElement('th');
-  nameEl.textContent = currentStore.name;
+  nameEl.textContent = this.name + ':';
   rowEl.appendChild(nameEl);
 
-
+//--for loop to produce cph stats for each time slot for this row (it is inside the store loop so it will go 16 times five different times....)
   for(var j = 0; j < timeSlot.times.length ; j++){
+    var storeRandomCPH = this.getAvgCookieCount();
```

Comment:
This works great for now, but in the long term, you may want to decouple the process of generating your numbers from the process of drawing them into the DOM.